db/src/db.rs

```diff
@@ -0,0 +1,254 @@
+   r#"CREATE INDEX idx_transactions_tx ON transactions (tx, added)"#,
+
+   // Fulltext indexing.
+   // A fulltext indexed value v is an integer rowid referencing fulltext_values.
+
+   // Optional settings:
+   // tokenize="porter"#,
+   // prefix='2,3'
+   // By default we use Unicode-aware tokenizing (particularly for case folding), but preserve
+   // diacritics.
+   r#"CREATE VIRTUAL TABLE fulltext_values
+        USING FTS4 (text NOT NULL, searchid INT, tokenize=unicode61 "remove_diacritics=0")"#,
+
+   // This combination of view and triggers allows you to transparently
+   // update-or-insert into FTS. Just INSERT INTO fulltext_values_view (text, searchid).
+   r#"CREATE VIEW fulltext_values_view AS SELECT * FROM fulltext_values"#,
+   r#"CREATE TRIGGER replace_fulltext_searchid
+        INSTEAD OF INSERT ON fulltext_values_view
+        WHEN EXISTS (SELECT 1 FROM fulltext_values WHERE text = new.text)
+        BEGIN
+          UPDATE fulltext_values SET searchid = new.searchid WHERE text = new.text;
+        END"#,
+   r#"CREATE TRIGGER insert_fulltext_searchid
+        INSTEAD OF INSERT ON fulltext_values_view
+        WHEN NOT EXISTS (SELECT 1 FROM fulltext_values WHERE text = new.text)
+        BEGIN
+          INSERT INTO fulltext_values (text, searchid) VALUES (new.text, new.searchid);
+        END"#,
+
+   // A view transparently interpolating fulltext indexed values into the datom structure.
+   r#"CREATE VIEW fulltext_datoms AS
+        SELECT e, a, fulltext_values.text AS v, tx, value_type_tag, index_avet, index_vaet, index_fulltext, unique_value
+          FROM datoms, fulltext_values
+          WHERE datoms.index_fulltext IS NOT 0 AND datoms.v = fulltext_values.rowid"#,
+   
+   // A view transparently interpolating all entities (fulltext and non-fulltext) into the datom structure.
+   r#"CREATE VIEW all_datoms AS
+        SELECT e, a, v, tx, value_type_tag, index_avet, index_vaet, index_fulltext, unique_value
+          FROM datoms
+          WHERE index_fulltext IS 0
+        UNION ALL
+        SELECT e, a, v, tx, value_type_tag, index_avet, index_vaet, index_fulltext, unique_value
+          FROM fulltext_datoms"#,
+
+   // Materialized views of the schema.
+   r#"CREATE TABLE idents (ident TEXT NOT NULL PRIMARY KEY, entid INTEGER UNIQUE NOT NULL)"#,
+   r#"CREATE TABLE schema (ident TEXT NOT NULL, attr TEXT NOT NULL, value BLOB NOT NULL, value_type_tag SMALLINT NOT NULL,
+        FOREIGN KEY (ident) REFERENCES idents (ident))"#,
+   r#"CREATE INDEX idx_schema_unique ON schema (ident, attr, value, value_type_tag)"#,
+   r#"CREATE TABLE parts (part TEXT NOT NULL PRIMARY KEY, start INTEGER NOT NULL, idx INTEGER NOT NULL)"#,
+   ];
+
+const V2_STATEMENTS: [&'static str; 19] = V1_STATEMENTS;
+
+fn set_user_version(conn: &rusqlite::Connection, version: i32) -> Result<i32> {
```

Comment:
This is a good idea, so I've filed https://github.com/mozilla/mentat/issues/190 to track it.